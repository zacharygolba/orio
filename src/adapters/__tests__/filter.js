// @flow

import Filter from '../filter'
import intoIter from '../../utils/into-iter'

let iter

beforeEach(() => {
  iter = new Filter(intoIter([1, 2, 3]), value => value > 1)
})

test('#next()', () => {
  expect(iter.next()).toMatchSnapshot()
  expect(iter.next()).toMatchSnapshot()
  expect(iter.next()).toMatchSnapshot()
  expect(iter.next()).toMatchSnapshot()
})

test('#sizeHint()', () => {
  expect(iter.sizeHint()).toEqual(3)
})
