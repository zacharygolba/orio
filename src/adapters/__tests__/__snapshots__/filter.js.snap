// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#next() 1`] = `
Object {
  "done": false,
  "value": 2,
}
`;

exports[`#next() 2`] = `
Object {
  "done": false,
  "value": 3,
}
`;

exports[`#next() 3`] = `
Object {
  "done": true,
  "value": undefined,
}
`;

exports[`#next() 4`] = `
Object {
  "done": true,
  "value": undefined,
}
`;
